src/nspawn/nspawn-network.c

```diff
@@ -743,3 +778,40 @@ int remove_veth_links(const char *primary, char **pairs) {
 
         return 0;
 }
+
+int network_iface_pair_parse_and_test(const char* iftype, char ***l, const char *p) {
+        _cleanup_free_ char *a = NULL, *b = NULL;
+        int r;
+
+        r = extract_first_word(&p, &a, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+        if (r < 0)
+                return log_error_errno(r, "Failed to extract first word in %s parameter: %m", iftype);
+        if (r == 0 || !ifname_valid(a))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "%s, interface name not valid: %s", iftype, a);
+
+        r = test_network_interface_initialized(a);
+        if (r < 0)
+                return r;
+
+        r = extract_first_word(&p, &b, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+        if (r < 0)
+                return log_error_errno(r, "Failed to extract second word in %s parameter: %m", iftype);
+        if (r == 0 || !ifname_valid(b)) {
+                free(b);
+                b = strdup(IFNAME_NOTSET);
+                if (!b)
+                        return log_oom();
+        }
+
+        if (p)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                        "%s, parameter not valid (garbage at the end: %s): %s", iftype, p, a);
+
+        r = strv_push_pair(l, a, b);
+        if (r < 0)
+                return log_error_errno(r, "Failed to push pair from %s parsed parameter: %m", iftype);
```

Comment:
this can only fail in case of oom, hence maybe just use `log_oom()` here, instead of making up your own message